### DISCUSSION TITLE
Send Mongo logs to Loki with OTEL

### DISCUSSION THREAD

**Discussion by nine-2-five** [microsoft/aspire#6977] [Q&A] [answered]
Should this be enough to send the Mongo logs to OTEL collector which is known to successfully send the logs of the API service to Loki?
Currently there are no log entries in Loki or Aspire Dashboard. But I see the entries for the API service project.
There are no Mongo log entries in console logs pertaining to OTEL, but the ENV variables are set in the Mongo container.

Telnet tests from Mongo to OTEL collector container are successful: `telnet otelcollector 4317`

```
OTEL_BSP_SCHEDULE_DELAY=1000
OTEL_EXPORTER_OTLP_PROTOCOL=grpc
OTEL_RESOURCE_ATTRIBUTES=service.instance.id=d38826dd
OTEL_EXPORTER_OTLP_ENDPOINT=http://otelcollector:4317
OTEL_TRACES_SAMPLER=always_on
OTEL_METRICS_EXEMPLAR_FILTER=trace_based
OTEL_SERVICE_NAME=mongo
OTEL_METRIC_EXPORT_INTERVAL=1000
OTEL_BLRP_SCHEDULE_DELAY=1000
```

```
var mongo = builder.AddMongoDB("mongo")
    .WithDataVolume()
    .WithOtlpExporter() // is this enough?
    .WithHttpEndpoint(port: 27090, targetPort: 27017)
    .WithLifetime(ContainerLifetime.Persistent);    
var mongodb = mongo.AddDatabase("mongodb");
```

**Comment by davidfowl** [comment]
This requires knowledge of mongo and how it sends telemetry to an otlp endpoint. Usual suspects are issues with https. Try to find documentation of a working example of this with docker compose.

**Reply by nine-2-five** [reply]
But then, what does the `.WithOtlpExporter()` do? This page states some form of automagicry: [https://learn.microsoft.com/en-us/dotnet/aspire/database/mongodb-integration?tabs=dotnet-cli#observability-and-telemetry](https://learn.microsoft.com/en-us/dotnet/aspire/database/mongodb-integration?tabs=dotnet-cli#observability-and-telemetry)

**Reply by davidfowl** [reply, marked as answer]
You can look at the souce code:

https://github.com/dotnet/aspire/blob/89e329c95ebac571567a58f27bfeb462d8e6b35d/src/Aspire.Hosting/OtlpConfigurationExtensions.cs#L117

It sets environment variables.
